### PR TITLE
add a test for channel +n / -n

### PR DESCRIPTION
**Pull request by slingamn**
This is broken in Ergo's master apparently :-)

### CODE REVIEW

**Comment by slingamn** [comment]
Ergo patch: https://github.com/ergochat/ergo/pull/2064

File: irctest/server_tests/chmodes/nooutside.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
+Channel "no outside" mode (`RFC 1459
```
Comment (progval) [inline irctest/server_tests/chmodes/nooutside.py:2]:
```suggestion
Channel "no external messages" mode (`RFC 1459
```

File: irctest/server_tests/chmodes/nooutside.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+<https://datatracker.ietf.org/doc/html/rfc1459#section-4.2.3.1>`__,
+`Modern <https://modern.ircdocs.horse/#no-external-messages-mode>`__)
+"""
+
+from irctest import cases
+from irctest.numerics import ERR_CANNOTSENDTOCHAN
+
+
+class NoOutsideTestCase(cases.BaseServerTestCase):
+    @cases.mark_specifications("RFC1459")
```
Comment (progval) [inline irctest/server_tests/chmodes/nooutside.py:12]:
```suggestion
    @cases.mark_specifications("RFC1459", "Modern")
```

File: irctest/server_tests/chmodes/nooutside.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        replies = self.getMessages("baz")
+        reply_cmds = {reply.command for reply in replies}
+        self.assertIn(ERR_CANNOTSENDTOCHAN, reply_cmds)
+        self.assertEqual(self.getMessages("chanop"), [])
+
+        # set the channel to -n: baz should be able to send now
+        self.sendLine("chanop", "MODE #chan -n")
+        replies = self.getMessages("chanop")
+        modeLines = [line for line in replies if line.command == "MODE"]
+        self.assertMessageMatch(modeLines[0], command="MODE", params=["#chan", "-n"])
```
Comment (progval) [inline irctest/server_tests/chmodes/nooutside.py:32]:
```suggestion
        self.assertMessageMatch(self.getMessage("chanop"), command="MODE", params=["#chan", "-n"])
```

File: irctest/server_tests/chmodes/nooutside.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        self.joinChannel("chanop", "#chan")
+        self.sendLine("chanop", "MODE #chan +n")
+        self.getMessages("chanop")
+
+        self.connectClient("baz", name="baz")
+        # this message should be suppressed completely by +n
+        self.sendLine("baz", "PRIVMSG #chan :hi from baz")
+        replies = self.getMessages("baz")
+        reply_cmds = {reply.command for reply in replies}
+        self.assertIn(ERR_CANNOTSENDTOCHAN, reply_cmds)
```
Comment (progval) [inline irctest/server_tests/chmodes/nooutside.py:25]: I'd be surprised if the CI passes for this one

Comment (progval) [reply irctest/server_tests/chmodes/nooutside.py:25]: oh well, I guess it does pass for once.

File: irctest/server_tests/chmodes/nooutside.py (hunk 3 above)
Comment (slingamn) [reply irctest/server_tests/chmodes/nooutside.py:32]: The original seems more flexible here?

File: irctest/server_tests/chmodes/nooutside.py (hunk 4 above)
Comment (slingamn) [reply irctest/server_tests/chmodes/nooutside.py:25]: This is largely copied and pasted from the test for `+m`.

File: irctest/server_tests/chmodes/nooutside.py (hunk 3 above)
Comment (progval) [reply irctest/server_tests/chmodes/nooutside.py]: there's no reason to allow non-MODE messages here

**Review by progval — APPROVED** [review]

**Pull request merged**